### DISCUSSION TITLE
Does the use of `np.random.RandomState` compromise randomness when used in multiprocessed PyTorch DataLoader?

### DISCUSSION THREAD

**Discussion by Antymon** [Project-MONAI/MONAI#2021] [General]
I have noticed that the PyTorch datasets implementations leverage `np.random.RandomState`.

I do wonder whether this might result in compromising pseudo-randomness in multiprocess scenarios:

> PyTorch uses multiprocessing to load data in parallel. The worker processes are created using the fork start method. This means each worker process inherits all resources of the parent, including the state of NumPy’s random number generator.

A detailed explanation of the problem I have in mind is available [here](https://tanelp.github.io/posts/a-bug-that-plagues-thousands-of-open-source-ml-projects/).

**Comment by Nic-Ma** [comment]
Hi @Antymon ,

I am not sure whether this question is within MONAI scope, but we already enhanced the random logic of DataLoader:
https://github.com/Project-MONAI/MONAI/blob/9f4da6acded249bba24c85eaee4ece256ed45815/monai/data/utils.py#L397

Thanks.
